pkg/controller/jobframework/webhook/defaulter.go

```diff
@@ -58,3 +63,70 @@ func (h *losslessDefaulter) Handle(ctx context.Context, req admission.Request) a
 	}
 	return response
 }
+
+func fieldExistsByJSONPath(object interface{}, path string) bool {
+	pathParts := strings.Split(path, "/")
+	// Invalid path. For more information, see https://jsonpatch.com/.
+	if len(pathParts) < 2 || len(pathParts) > 0 && pathParts[0] != "" {
+		return false
+	}
+	return fieldExistsByJSONPathHelper(object, pathParts[1:])
+}
+
+func fieldExistsByJSONPathHelper(object interface{}, pathParts []string) bool {
+	if object == nil {
+		return false
+	}
+
+	t := reflect.TypeOf(object)
+	v := reflect.ValueOf(object)
+
+	if v.Kind() == reflect.Pointer {
+		t = t.Elem()
+		v = v.Elem()
+	}
+
+	if t.Kind() != reflect.Struct {
+		return false
+	}
+
+	for i := 0; i < t.NumField(); i++ {
+		ft := t.Field(i)
+		fv := v.Field(i)
+
+		tag := ft.Tag.Get("json")
+		if tag == "" {
+			tag = ft.Name
+		} else if tagParts := strings.Split(tag, ","); len(tagParts) > 1 {
+			tag = tagParts[0]
```

Comment:
Can you check if it is legal to have `json:"bar , omitempty"` - with spaces. If legal, then we should trim but I'm not sure.

Reply:
Added separate function to get field name.